pkg/log/log.go

```diff
@@ -14,196 +14,111 @@
-	case "info":
-		return log.InfoLevel
-	}
-	return defaultLogLevel
+// SetLevel modifies the log level of the global logger. Returns the previous
+// level.
+func SetLevel(level zapcore.Level) zapcore.Level {
+	oldLevel := appLevel.Level()
+	appLevel.SetLevel(level)
+	return oldLevel
 }
 
-// modifyHook injects file name and line pos into log entry.
-type contextHook struct{}
-
-// Fire implements logrus.Hook interface
-// https://github.com/sirupsen/logrus/issues/63
-func (hook *contextHook) Fire(entry *log.Entry) error {
-	// these two num are set by manually testing
-	pc := make([]uintptr, 4)
-	cnt := runtime.Callers(10, pc)
-
-	for i := 0; i < cnt; i++ {
-		fu := runtime.FuncForPC(pc[i] - 1)
-		name := fu.Name()
-		if !isSkippedPackageName(name) {
-			file, line := fu.FileLine(pc[i] - 1)
-			entry.Data["file"] = path.Base(file)
-			entry.Data["line"] = line
-			break
-		}
+// ShortError contructs a field which only records the error message without the
+// verbose text (i.e. excludes the stack trace).
+//
+// In DM, all errors are almost always propagated back to `main()` where
+// the error stack is written. Including the stack in the middle thus usually
+// just repeats known information. You should almost always use `ShortError`
+// instead of `zap.Error`, unless the error is no longer propagated upwards.
+func ShortError(err error) zap.Field {
+	if err == nil {
+		return zap.Skip()
 	}
-	return nil
-}
-
-// Levels implements logrus.Hook interface.
-func (hook *contextHook) Levels() []log.Level {
-	return log.AllLevels
-}
-
-// isSKippedPackageName tests wether path name is on log library calling stack.
-func isSkippedPackageName(name string) bool {
-	return strings.Contains(name, "github.com/sirupsen/logrus") ||
-		strings.Contains(name, "github.com/pingcap/dm/pkg/log")
+	return zap.String("error", err.Error())
 }
 
-// textFormatter is for compatibility with ngaut/log
-type textFormatter struct {
-	DisableTimestamp bool
-	EnableEntryOrder bool
-}
-
-// Format implements logrus.Formatter
-func (f *textFormatter) Format(entry *log.Entry) ([]byte, error) {
-	var b *bytes.Buffer
-	if entry.Buffer != nil {
-		b = entry.Buffer
-	} else {
-		b = &bytes.Buffer{}
-	}
-
-	if !f.DisableTimestamp {
-		fmt.Fprintf(b, "%s ", entry.Time.Format(defaultLogTimeFormat))
-	}
-	if file, ok := entry.Data["file"]; ok {
-		fmt.Fprintf(b, "%s:%v:", file, entry.Data["line"])
-	}
-	fmt.Fprintf(b, " [%s] %s", entry.Level.String(), entry.Message)
-
-	if f.EnableEntryOrder {
-		keys := make([]string, 0, len(entry.Data))
-		for k := range entry.Data {
-			if k != "file" && k != "line" {
-				keys = append(keys, k)
-			}
-		}
-		sort.Strings(keys)
-		for _, k := range keys {
-			fmt.Fprintf(b, " %v=%v", k, entry.Data[k])
-		}
-	} else {
-		for k, v := range entry.Data {
-			if k != "file" && k != "line" {
-				fmt.Fprintf(b, " %v=%v", k, v)
-			}
-		}
-	}
-
-	b.WriteByte('\n')
-
-	return b.Bytes(), nil
+// L returns the current logger for Lightning.
```

Comment:
🙃

Reply:
really don't want to import lightning repo, it would be nice if It was in a public library.

Reply:
Yes we could copy the common parts into tidb-tools